gen_redirects.py

```diff
@@ -52,12 +52,6 @@
 # Untemplated redirects
 REDIRECTS = {
     "/api/": "/reference/",
-    "/api/backends/": "/reference/backends/",
```

Comment:
I think we might want to leave this one in (so that now-very-old links redirect to the support matrix)

Reply:
I think `"/reference/backends/"` isn't a valid target? https://ibis-project.org/reference/backends

Maybe it should just point to `"/backends/"` instead?

Reply:
Alternatively, I could add an `"/reference/backends/": "/backends/"` redirect?

Reply:
Maybe add the additional redirect, yeah.  I think that should be good.

Reply:
OK, pretty much just done that.